### ISSUE TITLE
Macro default variable that is a boolean type fails

### ISSUE THREAD

**Issue by Alex-Monahan** [duckdb/duckdb#13104]
### What happens?

Howdy! This is a super tiny issue...

If a macro default variable is a boolean type, the macro will not successfully create. This may be caused by automatically re-writing the literal `true` into `cast('t' as boolean)`.

This is a tiny issue since integer values will perform adequately. The main downside is that the parameters are not as "self-documenting" that there are only 2 possible values.

### To Reproduce

This fails:
```sql
CREATE OR REPLACE MACRO default_bool_test(param1:=true) AS 
    CASE WHEN param1 THEN 'Tis true' ELSE 'Nope' END 
```
```
Parser Error: Invalid parameter: 'CAST('t' AS BOOLEAN)'
```

This succeeds:
```sql
CREATE OR REPLACE MACRO default_bool_test(param1:=1) AS 
    CASE WHEN param1 THEN 'Tis true' ELSE 'Nope' END 
```

### OS:

MacOS

### DuckDB Version:

v1.0.1-dev2074

### DuckDB Client:

Python

### Full Name:

Alex Monahan

### Affiliation:

DuckDB Labs and MotherDuck

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a nightly build

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by flashmouse** [comment]
Could we add logic deal with `ExpressionType::OPERATOR_CAST` in `Transformer::TransformMacroFunction` during transform phase to solve this issue.

**Comment by lnkuiper** [comment]
Closed via #13317

**Comment by Alex-Monahan** [comment]
Thank you!!
